### PR TITLE
Deprecate the `SeedKubeScheduler` feature gate

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area auto-scaling
/kind cleanup

**What this PR does / why we need it**:
This PR deprecates the `SeedKubeScheduler` feature gate in favor of the `bin-packing` scheduling profile that can be configured for a Shoot referred by a ManagedSeed. See https://github.com/gardener/gardener/blob/v1.54.0/docs/usage/shoot_scheduling_profiles.md
 
For a gardenlet that enabled the feature gate the following warning log is printed (on startup):
```
W0829 18:11:26.449327   42201 feature_gate.go:235] Setting deprecated feature gate SeedKubeScheduler=true. It will be removed in a future release.
```

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/4741

**Release note**:

```other operator
gardenlet's `SeedKubeScheduler` feature gate is now deprecated in favor of the `bin-packing` scheduling profile that can be configured for a Shoot referred by a ManagedSeed.
```

### CODE REVIEW

File: docs/deployment/feature_gates.md [hunk 1]
```diff
@@ -46,61 +45,63 @@ The following tables are a summary of the feature gates that you can set on diff
...
+| WorkerPoolKubernetesVersion                  |         | `Removed`    | `1.52` |        |
+| DisableDNSProviderManagement                 | `false` | `Alpha`      | `1.41` | `1.49` |
+| DisableDNSProviderManagement                 | `true`  | `Beta`       | `1.50` | `1.51` |
+| DisableDNSProviderManagement                 | `true`  | `GA`         | `1.52` |        |
+| SecretBindingProviderValidation              | `false` | `Alpha`      | `1.38` | `1.50` |
+| SecretBindingProviderValidation              | `true`  | `Beta`       | `1.51` | `1.52` |
+| SecretBindingProviderValidation              | `true`  | `GA`         | `1.53` | `1.54` |
+| SecretBindingProviderValidation              |         | `Removed`    | `1.55` |        |
+| SeedKubeScheduler                            | `false` | `Alpha`      | `1.15` | `1.54` |
+| SeedKubeScheduler                            | `false` | `Deprecated` | `1.55` |        |
```
Comment (acumino) [inline docs/deployment/feature_gates.md:104]: This is not in line with lines 82-83. We should use only one way to list deprecated feature gates.

File: pkg/features/features.go [hunk 2]
```diff
@@ -135,7 +138,7 @@ var allFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	ManagedIstio:       {Default: true, PreRelease: featuregate.Beta},
 	APIServerSNI:       {Default: true, PreRelease: featuregate.Beta},
 	SeedChange:         {Default: true, PreRelease: featuregate.Beta},
-	SeedKubeScheduler:  {Default: false, PreRelease: featuregate.Alpha},
+	SeedKubeScheduler:  {Default: false, PreRelease: featuregate.Deprecated},
```
Comment (acumino) [inline pkg/features/features.go:141]: I guess this feature gate can be removed from example yamls as it is deprecated and not recommended to be used.

Comment (ialidzhikov) [reply pkg/features/features.go:141]: Yes, makes sense!

File: docs/deployment/feature_gates.md (hunk 1 above)
Comment (ialidzhikov) [reply docs/deployment/feature_gates.md:104]: Yes, I also noticed this. I think that the deprecations of `RotateSSHKeypairOnMaintenance` (https://github.com/gardener/gardener/pull/5985) or `ManagedIstio` and `APIServerSNI` (https://github.com/gardener/gardener/pull/6007) are not the most proper ones because they don't update the stage of the feature gate to be `Deprecated`. The corresponding feature gates stay in `Beta` and only have a doc string that they are deprecated. The `ManagedIstio` and `APIServerSNI` case is also different because when you have something deprecated it means "don't enable it, it will be removed" but we rather support and suggest enabling `ManagedIstio` and `APIServerSNI` despite the fact that they are deprecated.

Long story short, I decided to use the `Deprecated` stage instead of only adding a `(deprecated)` next to the feature gate name. I believe that this should be the proper way of deprecating a feature gate. For reference, example for deprecating a feature gate in K8s https://github.com/kubernetes/kubernetes/pull/102966 - they also used the `Deprecated` stage.

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by acumino — APPROVED** [review]
